src/main/java/org/folio/service/impl/RequestServiceImpl.java

```diff
@@ -71,17 +98,87 @@ public RequestWrapper createSecondaryRequest(Request request, String borrowingTe
     throw new RequestCreatingException(errorMessage);
   }
 
-  private RequestWrapper createSecondaryRequest(Request request, String lendingTenantId) {
-    final String requestId = request.getId();
-    log.info("createSecondaryRequest:: creating secondary request {} in lending tenant {}",
-      requestId, lendingTenantId);
-    Request secondaryRequest = tenantScopedExecutionService.execute(lendingTenantId,
-      () -> circulationClient.createInstanceRequest(request));
-    log.info("createSecondaryRequest:: secondary request {} created in lending tenant {}",
-      requestId, lendingTenantId);
-    log.debug("createSecondaryRequest:: secondary request: {}", () -> secondaryRequest);
-
-    return new RequestWrapper(secondaryRequest, lendingTenantId);
+  public User getOrCreateUser(User user) {
+    return getOrCreate(user.getId(), user, userService::find, userService::create);
+  }
+
+  public ServicePoint getOrCreateServicePoint(ServicePoint servicePoint) {
+    return getOrCreate(servicePoint.getId(), servicePoint, servicePointService::find,
+      servicePointService::create);
+  }
+
+  public <T> T getOrCreate(String objectId, T object, Function<String, T> finder,
```

Comment:
Should we move to some common parent, to use in other services, for example in UserService?- Like BaseService